### PR TITLE
Fix for ANDROID build - (OpenSSL, JAVA) on master-branch

### PR DESCRIPTION
**Pull request by Werner86**
When trying to do a fresh build of the master-branch for TARGET=ANDROID it currently fails because of two reasons:

**OpenSSL (libs.py) **
The file 
https://www.openssl.org/source/openssl-3.0.0-alpha4.tar.gz
has been moved to
https://www.openssl.org/source/ **old/3.0/** openssl-3.0.0-alpha4.tar.gz
Therefore the build breaks if this file has not been downloaded upfront / manually.

**JAVA (android.mk)**
The JAVA-build for android fails on newer JDKs because building JAVA 1.6 is not supported anymore. Therefore changed the build to 1.7

### CODE REVIEW

File: build/python/build/libs.py [hunk 1]
```diff
@@ -54,8 +54,8 @@
 )
 
 openssl = OpenSSLProject(
-    'https://www.openssl.org/source/openssl-3.0.0-alpha4.tar.gz',
-    'ftp://ftp.cert.dfn.de/pub/tools/net/openssl/source/openssl-3.0.0-alpha4.tar.gz',
+    'https://www.openssl.org/source/old/3.0/openssl-3.0.0-alpha4.tar.gz',
+    'ftp://ftp.cert.dfn.de/pub/tools/net/openssl/source/old/3.0/openssl-3.0.0-alpha4.tar.gz',
```
Comment (MaxKellermann) [inline build/python/build/libs.py:58]: I'd prefer updating to the latest version.

Comment (Werner86) [reply build/python/build/libs.py:58]: i agree but the newest version is not working without adjustments so we should have a working built at first or?

Comment (Werner86) [reply build/python/build/libs.py]: the commit-messages have been separated for the two different changes

**Comment by MaxKellermann** [comment]
I just wanted to cherry-pick the Java 1.7 change, but it has a bad commit message. It doesn't explain why this change was done.

**Comment by Werner86** [comment]
> I just wanted to cherry-pick the Java 1.7 change, but it has a bad commit message. It doesn't explain why this change was done.

you're right... was a bit too short - now it should be fine

**Comment by MaxKellermann** [comment]
Java commit cherry-picked to v6.8.x: 4c369402dad57dd739c4974d3b60eb3e5aecce6a

**Comment by Werner86** [comment]
> I'd prefer updating to the latest version.

There is currently an incompatability between the latest version of openssl and curl. In the latest alpha-version of openssl there has been removed the function SSL_get_peer_certificate (deprecated with version 3.0.0)
https://www.openssl.org/docs/manmaster/man3/SSL_get_peer_certificate.html
which is needed by curl (even in the latest version).
I recommend to either step back to a release-version of openssl or use the "old" version openssl-3.0.0-alpha4 which is in my commit.

**Comment by MaxKellermann** [comment]
de746942d1e4d7a31c48a6a7eaa887d2694b2aca

**Pull request closed without merging**